File: Jackfruit.CSharpShim/ShimGenerator.cs
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.Text;
using System.Text;

namespace Generator.CSharpShim
{
    [Generator]
    public class CustomGenerator : ISourceGenerator
    {
        public void Initialize(GeneratorInitializationContext context) { }

        public void Execute(GeneratorExecutionContext context)
        {
            context.AddSource("myGeneratedFile.cs", SourceText.From(@"
namespace GeneratedNamespace
{
    public class GeneratedClass
        }
    }
}", Encoding.UTF8));
        }
    }
}

File: JackfruitAppModel/ShimGenerator.fs
namespace Generator.CSharpShim

open Microsoft.CodeAnalysis;
open Microsoft.CodeAnalysis.Text;
open System.Text;

[<Generator>]
type CustomGenerator() =
    interface ISourceGenerator with
        member this.Initialize (context: GeneratorInitializationContext) =
            ()

        member this.Execute (context: GeneratorExecutionContext) =
            context.AddSource("myGeneratedFile.cs", SourceText.From(@"
    namespace GeneratedNamespace
    {
    public class GeneratedClass
        {
        }
    }", Encoding.UTF8))
